UserManagement/accounts/presentation/views.py

```diff
@@ -31,7 +30,44 @@ def update(self, request, *args, **kwargs):
     
 
 class InstructorRateViewSet(viewsets.ModelViewSet):
-    permission_classes = [HasViewInstructorRateClaim]
     queryset = InstructorRate.objects.all()
     serializer_class = InstructorRateSerializer
+    
+    def get_permissions(self):
+        if self.action == 'create':
+            permission_classes = [CanCreateInstructorRate]
+        elif self.action == 'destroy':
+            permission_classes = [CanDeleteInstructorRateAdminOnly]
+        else:
+            permission_classes = [HasViewInstructorRateClaim]
+        return [permission() for permission in permission_classes]
+    
+    def create(self, request, *args, **kwargs):
+        # Check if the user is authenticated
+        if not request.user.is_authenticated:
+            return Response({'error': 'Authentication required'}, status=status.HTTP_401_UNAUTHORIZED)
+    
+        # Check if the user has the necessary claim
+        if CreateInstructorRate not in request.user.claims:
+            return Response({'error': 'You do not have permission'}, status=status.HTTP_403_FORBIDDEN)
```

Comment:
"do not have permission" for what ?
consider extending the response with more precise information

Reply:
what Vitalik think ?

Reply:
Agree with Nazar, not enough descriptive